Add support for RISC-V

chazapis opened the pull request:
<!--
PLEASE READ THIS MESSAGE.

Documentation fixes or enhancements:
- for Traefik v2: use branch v2.10
- for Traefik v3: use branch v3.0

Bug fixes:
- for Traefik v2: use branch v2.10
- for Traefik v3: use branch v3.0

Enhancements:
- for Traefik v2: we only accept bug fixes
- for Traefik v3: use branch master

HOW TO WRITE A GOOD PULL REQUEST? https://doc.traefik.io/traefik/contributing/submitting-pull-requests/

-->

### What does this PR do?

<!-- A brief description of the change being made with this pull request. -->

This PR adds RISC-V support, by generating a RISC-V binary.

### Motivation

<!-- What inspired you to submit this pull request? -->

The work is part of an ongoing effort to run [K3s on RISC-V](https://github.com/k3s-io/k3s/pull/7778). While the main functionality is there, we also need supporting tools and utilities working, such as Traefik which is deployed by default.

### More

- [ ] Added/updated tests
- [ ] Added/updated documentation

### Additional Notes

<!-- Anything else we should know when reviewing? -->

There are no code changes in this PR. Just the `riscv64` added to the GoReleaser configuration.

ldez commented:
Hello,

can you rebase your PR onto the v2.10 branch?

Note: the PR from GitHUb organization are a problem for our automation because our bot or us cannot edited the PR.
I recommend using a personal fork.

chazapis commented:
> Hello,
> 
> can you rebase your PR onto the v2.10 branch?
> 
> ```
> git rebase --onto=upstream/v2.10 60bc47d00
> ```
> 
> Note: PRs from GitHub organization are a problem for our automation because our bot and us cannot edit the PR. I recommend using a personal fork.

Rebased and changed the destination of this PR to `v2.10`.

Regarding that this PR is from an organization: Could I probably add some permissions to the repository to solve this? Otherwise should I close this PR and open another one from my personal account?

ldez commented:
> Otherwise should I close this PR and open another one from my personal account?

This is the best solution for us.

chazapis commented:
Closing this in favor of #10026.
